### PR TITLE
Add migration guide `0.x` to `1.0.0`

### PR DESCRIPTION
**Pull request by dislick**
Add migration guide to  `/docs`, as discussed in #348.

### CODE REVIEW

**Comment by aralroca** [comment]
Don't worry about the tests.

File: docs/migration-guide-1.0.0.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+6. Rename directory `pages_` to `pages`:
+
+```bash
+rm -rf pages
+mv pages_ pages
+```
+
+7. Make sure to update plural suffixes. More information on this can be found [in the README](https://github.com/vinissimus/next-translate/tree/1.0.0-experimental#5-plurals).
+
+```diff
```
Comment (aralroca) [inline docs/migration-guide-1.0.0.md:74]: The only breaking change is replacing `_plural` to `_other`. For numbers `_0`, `_999`, etc we still support them for **exact matches** with all the locales, which is how it worked before. For example:

```json
{
  "cart-message_0": "The cart is empty",
  "cart-message_one": "The cart has only {{count}} product",
  "cart-message_other": "The cart has {{count}} products",
  "cart-message_999": "The cart is full, you can't buy more products",
}
```

```

File: docs/migration-guide-1.0.0.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
+# Migration Guide `0.x` to `1.0.0`
+
+This migration guide describes how to upgrade existing projects using `next-translate@0.x` to `next-translate@1.0.0`.
+
+1. Update `next-translate` to `^1.0.0`. If you're using a recent version of `npm` you can run this command:
+
+```bash
+npm install next-translate@^1.0.0
```
Comment (aralroca) [inline docs/migration-guide-1.0.0.md:8]: it will be nice to add an example with yarn 😊

Comment (dislick) [reply docs/migration-guide-1.0.0.md:12]: Okay, I added the yarn example :)

File: docs/migration-guide-1.0.0.md (hunk 1 above)
Comment (dislick) [reply docs/migration-guide-1.0.0.md:78]: My bad, I just updated step 7 to only include the change from `_plural` to `_other`. Feel free to edit this yourself, I have little experience with plurals.

**Review by aralroca — APPROVED** [review]
Great work @dislick. Thank you so much! And welcome as first-time contributor!! 🔥

**Comment by aralroca** [comment]
@allcontributors please add @dislick for documentation

**Pull request merged**